Bisq-swap-3

chimp1984 opened the pull request.

chimp1984 committed:
Remove handling for bundledEnvelopes in send method

The code was never executed as it is not usual that on
one connection multiple messages are sent in a short period.
The handling for bundledEnvelopes has been implemented later
into the broadcast domain, there is does do its job.

chimp1984 committed:
Rename AtomicOfferPayload to BsqSwapOfferPayload

chimp1984 committed:
Add extraDataMap to BsqSwapOfferPayload

chimp1984 committed:
Rename AtomicAccountPayload to BsqSwapAccountPayload

Remove generic_string (was not used)

chimp1984 committed:
Rename AtomicAccount to BsqSwapAccount

chimp1984 committed:
Rename PaymentMethod.ATOMIC to PaymentMethod.BSQ_SWAP

chimp1984 committed:
Rename ProcessModelI to TradeProtocolModel

chimp1984 committed:
Rename AtomicProcessModel to BsqSwapProtocolModel

chimp1984 committed:
Rename AtomicProcessModel to BsqSwapProtocolModel in protobuf

chimp1984 committed:
Remove unused params

chimp1984 committed:
Rename AtomicTrade to BsqSwapTrade

chimp1984 committed:
Rename AtomicTrade to BsqSwapTrade in protobuf

chimp1984 committed:
Rename AtomicTradeManager to BsqSwapTradeManager

chimp1984 committed:
Move bisq.core.trade.atomic.AtomicTxBuilder to bisq.core.btc.AtomicTxBuilder

As it is about transaction domain I think it fits better in the bitcoin domain

chimp1984 committed:
Rename AtomicTxBuilder to BsqSwapTxHelper

Builder would suggest the builder pattern, so maybe
better use helper.

chimp1984 committed:
Rename AtomicMakerProtocol to BsqSwapMakerProtocol
and AtomicTakerProtocol to BsqSwapTakerProtocol

chimp1984 committed:
Move bisq.core.trade.protocol.BsqSwapProtocolModel to bisq.core.trade.atomic.protocol.BsqSwapProtocolModel

chimp1984 committed:
Move trade model classes to bisq.core.trade.model package

Keep bsqSwap and normal trade separate. Leave common classed
in base package.
No good name yet for default trade (default cannot be used as package name)

chimp1984 committed:
Move protocol classes to bisq.core.trade.protocol package

chimp1984 committed:
Move common classes level up

chimp1984 committed:
Rename ProcessModelServiceProvider to Provider

chimp1984 committed:
Move common classes to misc package

chimp1984 committed:
Move bisq.core.trade.protocol.messages to bisq.core.trade.messages

chimp1984 committed:
Move TradeManager, MakerTrade, TakerTrade one level up

Cleanups

chimp1984 committed:
Update protocol and protocol model classes

chimp1984 committed:
Remove Atomic prefixes for tasks

The package should be enough to qualify them as
BsqSwap tasks. They are only used in the limited scope
of the protocol package.

chimp1984 committed:
Rename CreateAtomicTxRequest and CreateAtomicTxResponse

chimp1984 committed:
Rename methods, fields with atomic prefix

chimp1984 committed:
Rename views

chimp1984 committed:
Rename AtomicOfferFunding

chimp1984 committed:
Refactor placeoffer packages

chimp1984 committed:
Replace TransactionResultHandler with Runnable

There is no tx set at bsqSwap place offer protocol

chimp1984 committed:
Replace ocurrances of atomic with bsqSwap

chimp1984 committed:
Use OfferPayloadBase instead of ProtectedStoragePayload as type

As we are in the offer domain we should use the domain relevant type.

chimp1984 committed:
Remove redundant OfferPayload.Direction

Move Direction one level up and rename to OfferDirection
to match protobuf definition

chimp1984 committed:
Move AvailabilityResult to bisq.core.offer.availability

sqrrm commented on core/src/main/java/bisq/core/trade/atomic/AtomicMakerTrade.java:
```diff
@@ -90,19 +90,17 @@ public static Tradable fromProto(protobuf.AtomicMakerTrade atomicMakerTradeProto
+                proto.getTakerFee(),
+                BsqSwapProtocolModel.fromProto(proto.getBsqSwapProtocolModel(), coreProtoResolver),
+                proto.getErrorMessage(),
+                State.fromProto(proto.getState())));
```
Indentation

sqrrm commented on core/src/main/java/bisq/core/trade/atomic/AtomicTakerTrade.java:
```diff
@@ -90,19 +90,17 @@ public static Tradable fromProto(protobuf.AtomicTakerTrade atomicTakerTradeProto
+                proto.getTakerFee(),
+                BsqSwapProtocolModel.fromProto(proto.getBsqSwapProtocolModel(), coreProtoResolver),
+                proto.getErrorMessage(),
+                State.fromProto(proto.getState())));
```
Indentation

sqrrm commented on core/src/main/java/bisq/core/offer/placeoffer/offer/tasks/ValidateOffer.java:
```diff
@@ -40,6 +41,9 @@ protected void run() {
             runInterceptHook();
 
+            checkArgument(offer.getOfferPayloadBase() instanceof OfferPayload,
+                    "OfferPayload must be OfferPayload");
```
Very helpful comment :)

chimp1984 replied:
Haha, find and replace artefacts....

sqrrm commented on proto/src/main/proto/pb.proto:
```diff
@@ -787,17 +787,11 @@ message MailboxStoragePayload {
     NodeAddress owner_node_address = 3;
     PubKeyRing pub_key_ring = 4;
-    Direction direction = 5;
+    OfferDirection direction = 5;
```
I think this will break backwards compatibility, but maybe it works since the enums are identical except for scope

chimp1984 replied:
We have to test it with backward comp. I hope it does not break.... but might be

chimp1984 committed:
Small refactorings, cleanups

chimp1984 committed:
Refactor BsqSwapWalletWatcher and OpenOfferManager

I think the dependency should be inverted here, so that the
OpenOfferManager does not know about the details about BsqSwap
funding.
Removed FundingListener, use listener on observable list instead
and maintain a filtered list in the BsqSwapWalletWatcher.

sqrrm commented on core/src/main/java/bisq/core/offer/BsqSwapWalletWatcher.java:
```diff
@@ -24,86 +24,131 @@
-        if (!feeService.isFeeAvailable())
-            return false;
+    private void updateFundingState(OpenOffer openOffer) {
+        if (!feeService.isFeeAvailable()) {
+            // At startup the fee service might not be available still so we call again after a delay
+            UserThread.runAfter(() -> {
+                if (openBsqSwapOffers.contains(openOffer)) {
+                    updateFundingState(openOffer);
+                }
+            }, 10);
+            return;
+        }
```
Looks like this would create an infinite loop if `isFeeAvailable()` keeps being false.

chimp1984 replied:
Yes, it repeats as long the fees are not available after every 10 sec. I think thats ok, but could add some counter as well or increase delay with each iteration.

sqrrm replied:
It's doing it for every offer which is a bit unnecessary, would be better to have this loop not for each offer but just once for the instance

chimp1984 replied:
yes i added a counter to reduce dependency on fee service. we have default rate if service is failing

sqrrm reviewed (commented):
@chimp1984 could you look at the infinite loop comment?

I think the api tests have some issue but I will let that be for now, Stan said he can look at it later.

chimp1984 committed:
Add counter to fee service check

chimp1984 committed:
Fix copy paste error
